catroid/src/androidTest/java/org/catrobat/catroid/test/cloudmessaging/CustomFirebaseMessagingServiceTest.java

```diff
@@ -0,0 +1,80 @@
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.test.cloudmessaging;
+
+import android.app.Notification;
+import android.app.NotificationManager;
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+import android.support.v4.app.NotificationCompat;
+
+import org.catrobat.catroid.cloudmessaging.CloudMessage;
+import org.catrobat.catroid.cloudmessaging.CloudMessaging;
+import org.catrobat.catroid.cloudmessaging.NotificationBuilderProvider;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.HashMap;
+
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class CustomFirebaseMessagingServiceTest {
+
+	private NotificationBuilderProvider mockNotificationBuilderProvider;
+	private NotificationCompat.Builder mockNotificationBuilder;
+	private Notification mockNotification;
+	private NotificationManager mockNotificationManager;
+
+	@Before
+	public void setUp() {
+		mockNotificationBuilder = Mockito.mock(NotificationCompat.Builder.class, Mockito.RETURNS_SELF);
+		mockNotification = Mockito.mock(Notification.class);
+		mockNotificationBuilderProvider = Mockito.mock(NotificationBuilderProvider.class);
+		when(mockNotificationBuilderProvider.get()).thenReturn(mockNotificationBuilder);
+		when(mockNotificationBuilder.build()).thenReturn(mockNotification);
+		mockNotificationManager = Mockito.mock(NotificationManager.class);
+	}
+
+	@Test
+	public void testNotificationIsBuildAndShown() {
+		HashMap<String, String> data = new HashMap<>();
+		data.put(CloudMessage.TITLE, "Title");
+		data.put(CloudMessage.MESSAGE, "Message");
+		data.put(CloudMessage.WEB_PAGE_URL, "http://TEST.COM");
+
+		Context context = InstrumentationRegistry.getTargetContext();
+		CloudMessage cloudMessage = new CloudMessage(data);
+
+		CloudMessaging cloudMessaging = new CloudMessaging(context, mockNotificationBuilderProvider, mockNotificationManager);
+		cloudMessaging.showNotification(cloudMessage);
+
+		verify(mockNotificationBuilder).setContentTitle(eq("Title"));
+		verify(mockNotificationBuilder).setContentText(eq("Message"));
```

Comment:
icon, sound, etc are missing.
If the webPageUrl isn't testable, test that setContentIntent is called with a non-null value.